packages/client/src/service/skeleton.ts

```diff
@@ -1283,16 +1300,16 @@ export class Skeleton extends MetaDBManager {
           await this.deleteBlock(block)
         }
       })
-      if (this.fillLogIndex >= this.config.numBlocksPerIteration) {
+      if (fillLogIndex >= this.config.numBlocksPerIteration) {
         this.config.logger.debug(
           `Skeleton canonical chain fill status: canonicalHead=${canonicalHead} chainHead=${this.chain.blocks.height} subchainHead=${subchain.head}`
         )
-        this.fillLogIndex = 0
+        fillLogIndex = 0
       }
     }
     this.filling = false
     this.config.logger.debug(
-      `Successfully put blocks start=${start} end=${canonicalHead} skeletonHead=${subchain.head} from skeleton chain to canonical syncTargetHeight=${this.config.syncTargetHeight}`
+      `Successfully put=${fillLogIndex} skipped=${skippedLogIndex} blocks start=${start} end=${canonicalHead} skeletonHead=${subchain.head} from skeleton chain to canonical syncTargetHeight=${this.config.syncTargetHeight}`
```

Comment:
```suggestion
      `Successfully put=${fillLogIndex} skipped (because already inserted)=${skippedLogIndex} blocks start=${start} end=${canonicalHead} skeletonHead=${subchain.head} from skeleton chain to canonical syncTargetHeight=${this.config.syncTargetHeight}`
```

Is it true that we should never be able to put a block and then skip one higher than it, right?  So, if we're doing canonical fill, we might skip the first `x` blocks because some other process already put them in the chain, but we would then fill any remaining in this current iteration, right?

Reply:
yes, except that because of any race condition, it might not be just `first` x and can happen anywhere mid segment because of the async / concurrent nature of fcU calls